<!-- -*- mode: XML -*- -->
<launch>

  <!-- Optionally launch in GDB, for debugging -->
  <arg name="debug" default="false" />
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <!-- VESC driver parameters -->
  <arg name="node_name" value="vesc_driver_node" />
  <arg name="port" value="/dev/ttyACM0" />
  <arg name="frame_id" value="" />
  <arg name="control_mode" value="duty_cycle" /> <!-- control_mode = {duty_cycle, current, brake, speed, position} -->
  <arg name="feedback_mode" value="pid_position" /> <!-- feedback_mode = {duty_cycle, current, speed, pid_position, encoder_position} -->
  <arg name="servo_mode" value="off" /> <!-- servo_mode = {on, off} -->

  <!-- VESC driver node -->
  <node pkg="vesc_driver" type="vesc_driver_node" name="$(arg node_name)"
        output="screen" launch-prefix="$(arg launch_prefix)" >
    <param name="port" value="$(arg port)" />
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="control_mode" value="$(arg control_mode)" />
    <param name="feedback_mode" value="$(arg feedback_mode)" />
    <param name="servo_mode" value="$(arg servo_mode)" />
  </node>

</launch>
